src/crypto/equihash.cpp

```diff
@@ -514,8 +639,12 @@ bool Equihash<N,K>::OptimisedSolve(const eh_HashState& base_state,
             icv.reserve(recreate_size);
             for (eh_index j = 0; j < recreate_size; j++) {
                 eh_index newIndex { UntruncateIndex(partialSoln.get()[i], j, CollisionBitLength + 1) };
-                icv.emplace_back(base_state, newIndex, HashLength,
-                                 CollisionBitLength, CollisionByteLength);
+                if (j == 0 || newIndex % IndicesPerHashOutput == 0) {
```

Comment:
The *solution indices* are not contiguous. The above loop is creating the possible values for a *single* solution index. To give an example, say we had 4-bit indices truncated to 2 bits, and we had the truncated solution index `01`. There are four contiguous possibilities:

- `0100`
- `0101`
- `0110`
- `0111`

If we then have another truncated solution index `10`, there are a separate four contiguous possibilities, and so on.